source/taskmanager-client/src/main/java/nl/aerius/taskmanager/client/mq/RabbitMQWorkerMonitor.java

```diff
@@ -0,0 +1,185 @@
+    observers.remove(observer);
+  }
+
+  /**
+   * Start the watcher.
+   *
+   * @throws IOException
+   */
+  public void start() throws IOException {
+    final Connection c = factory.getConnection();
+    channel = c.createChannel();
+    channel.exchangeDeclare(AERIUS_EVENT_EXCHANGE, EXCHANGE_TYPE);
+    final String queue = channel.queueDeclare().getQueue();
+    channel.queueBind(queue, AERIUS_EVENT_EXCHANGE, "");
+
+    consumer = new DefaultConsumer(channel) {
+      @Override
+      public void handleDelivery(final String consumerTag, final Envelope envelope, final AMQP.BasicProperties properties, final byte[] body)
+          throws IOException {
+        RabbitMQWorkerMonitor.this.handleDelivery(properties);
+      }
+
+      @Override
+      public void handleShutdownSignal(final String consumerTag, final ShutdownSignalException sig) {
+        if (sig.isInitiatedByApplication()) {
+          LOG.info("Worker event monitor {} was shut down by the application.", consumerTag);
+        } else {
+          LOG.debug("Worker event monitor {} was shut down.", consumerTag);
+          // restart
+          try {
+            try {
+              channel.abort();
+            } catch (final IOException e) {
+              // Eat error when closing channel.
+            }
+            start();
+            LOG.info("Restarted worker event monitor ; {}", consumerTag);
+          } catch (final IOException e) {
+            LOG.debug("Worker event monitor restart failed", e);
+          }
+        }
+      }
+    };
+    channel.basicConsume(queue, true, consumer);
+  }
+
+  private void handleDelivery(final AMQP.BasicProperties properties) {
+    final Map<String, Object> headers = properties.getHeaders();
+    final String queueName = getParam(headers, HEADER_PARAM_QUEUE);
+    final int workerSize = getParamInt(headers, HEADER_PARAM_WORKER_SIZE, -1);
+    final int utilisationSize = getParamInt(headers, HEADER_PARAM_UTILISATION, -1);
+
+    observers.forEach(ro -> ro.updateWorkers(queueName, workerSize, utilisationSize));
+  }
+
+  /**
+   * Returns the value in the header parameter or null.
+   *
+   * @param headers headers map
+   * @param param param to return the value for
+   * @return value of param or null if not present
+   */
+  private String getParam(final Map<String, Object> headers, final String param) {
+    final Object value = headers.get(param);
+
+    return value == null ? null : value.toString();
+  }
+
+  /**
+   * Returns the int value in the header parameter or the other value.
+   *
+   * @param headers headers map
+   * @param param param to return the value for
+   * @param other if param is not present this value is returned
+   * @return int value of param or other if not present
+   */
+  private int getParamInt(final Map<String, Object> headers, final String param, final int other) {
+    final String value = getParam(headers, param);
+
+    return value == null ? other : Integer.valueOf(value);
```

Comment:
What happens when this isn't a proper int? While it should always be one, I think a little check might not hurt. Though it might be handled properly because of a generic try/catch somewhere higher.

Reply:
I've wrapped it in a try-catch